### PR TITLE
Stop building Bionic Workbench Docker Products

### PR DESCRIPTION
**Pull request by AndrewMcClain**
This addresses some sections of rstudio/rstudio-pro#4470

After the previous PR we decide to not introduce focal docker products, just remove the bionic and leave the jammy builds alone.

Removes bionic dockerfiles for workbench, r-session-complete and workbench-for-microsoft-azure-ml
Removes bionic builds from workflows
Changes default os version to 2204

### CODE REVIEW

**Review by ianpittwood — APPROVED** [review]
LGTM, but I'd still like to hear @colearendt 's opinion on disabling the build vs fully removing the definitions. I don't want to cause a problem if there happens to be customers that still rely on these definitions right now and haven't been able to upgrade yet.

File: .github/workflows/build-latest.yaml [hunk 1]
```diff
@@ -200,18 +200,16 @@ jobs:
...
           - {product: 'connect-content-init', os: 'ubuntu2204', r-primary: "4.2.0", r-alternate: "3.6.2", py-primary: "3.9.5", py-alternate: "3.8.10"}
           - {product: 'package-manager', os: 'ubuntu1804', r-primary: "4.2.0", r-alternate: "3.6.2", py-primary: "3.9.5", py-alternate: "3.8.10"}
           - {product: 'package-manager', os: 'ubuntu2204', r-primary: "4.2.0", r-alternate: "3.6.2", py-primary: "3.9.5", py-alternate: "3.8.10"}
-          - {product: 'r-session-complete', os: 'ubuntu1804', r-primary: "4.2.3", r-alternate: "4.1.3", py-primary: "3.9.14", py-alternate: "3.8.15"}
           - {product: 'r-session-complete', os: 'centos7', r-primary: "4.2.3", r-alternate: "4.1.3", py-primary: "3.9.14", py-alternate: "3.8.15"}
           - {product: 'r-session-complete', os: 'ubuntu2204', r-primary: "4.2.3", r-alternate: "4.1.3", py-primary: "3.9.14", py-alternate: "3.8.15"}
-          - {product: 'workbench-for-microsoft-azure-ml', os: 'ubuntu1804', r-primary: "4.2.3", r-alternate: "4.1.3", py-primary: "3.9.14", py-alternate: "3.8.15"}
```
Comment (colearendt) [inline .github/workflows/build-latest.yaml:214]: I don't know that we have thought through side-effects here. We should probably couple this PR with the Workbench 2023.06 release.

**Review by colearendt — COMMENTED** [review]
Generally, I like it! Well done chasing everything through. My only thought - do we want to move this into a `dev-pwb` branch, so we can pair w/ the release of 2023.06 (the first product release w/ no bionic support)?

File: .github/workflows/build-latest.yaml (hunk 1 above)
Comment (AndrewMcClain) [reply .github/workflows/build-latest.yaml:214]: Workbench 2023.06 is now out

**Comment by AndrewMcClain** [comment]
@colearendt @ianpittwood @cm421 @melissa-barca @atheriel 

Now with the release of 2023.06 We would like to get this merged asap. Any issue with that?

**Review by cm421 — APPROVED** [review]
LGTM!

**Review by colearendt — APPROVED** [review]
LGTM! Let's merge this with the release!

**Pull request merged**